### ISSUE TITLE
The attention mask and the pad token id were not set

### ISSUE THREAD

**Issue by 2152505** [manycore-research/SpatialLM#6]
![Image](https://github.com/user-attachments/assets/0a516279-852a-4131-900a-306ce190a750)

![Image](https://github.com/user-attachments/assets/de3a681c-5765-43f1-ad34-d21ed16f8c24)

**Comment by 2152505** [comment]
请问这个报错是为什么，我是按照readme文档进行的操作，是否是那一步出现了问题

**Comment by SamMaoYS** [comment]
可以问一下你的环境配置吗，我们自己test的时候还没有遇到过类似的问题。

**Comment by 2152505** [comment]
我是按照文档里进行配置的，这是环境的配置文件

name: spatiallm
channels:
  - conda-forge
  - nvidia/label/cuda-12.4.0
  - https://mirrors.tuna.tsinghua.edu.cn/anaconda/cloud/conda-forge
  - https://mirrors.tuna.tsinghua.edu.cn/anaconda/cloud/conda-forge/
  - https://mirrors.tuna.tsinghua.edu.cn/anaconda/pkgs/free/
  - https://mirrors.tuna.tsinghua.edu.cn/anaconda/pkgs/main/
  - https://mirrors.tuna.tsinghua.edu.cn/anaconda/cloud/pytorch/
  - https://mirrors.tuna.tsinghua.edu.cn/anaconda/cloud/msys2/
dependencies:
  - _libgcc_mutex=0.1=conda_forge
  - _openmp_mutex=4.5=2_gnu
  - alsa-lib=1.2.13=hb9d3cd8_0
  - attr=2.5.1=h166bdaf_1
  - binutils=2.43=h4852527_4
  - binutils_impl_linux-64=2.43=h4bf12b8_4
  - binutils_linux-64=2.43=h4852527_4
  - bzip2=1.0.8=h4bc722e_7
  - c-compiler=1.9.0=h2b85faf_0
  - ca-certificates=2025.1.31=hbcca054_0
  - cuda-cccl_linux-64=12.8.90=ha770c72_1
  - cuda-command-line-tools=12.8.1=ha770c72_0
  - cuda-compiler=12.8.1=hbad6d8a_0
  - cuda-crt-dev_linux-64=12.8.93=ha770c72_1
  - cuda-crt-tools=12.8.93=ha770c72_1
  - cuda-cudart=12.8.90=h5888daf_1
  - cuda-cudart-dev=12.8.90=h5888daf_1
  - cuda-cudart-dev_linux-64=12.8.90=h3f2d84a_1
  - cuda-cudart-static=12.8.90=h5888daf_1
  - cuda-cudart-static_linux-64=12.8.90=h3f2d84a_1
  - cuda-cudart_linux-64=12.8.90=h3f2d84a_1
  - cuda-cuobjdump=12.8.90=hbd13f7d_1
  - cuda-cupti=12.8.90=hbd13f7d_0
  - cuda-cupti-dev=12.8.90=h5888daf_0
  - cuda-cuxxfilt=12.8.90=hbd13f7d_1
  - cuda-documentation=12.4.99=0
  - cuda-driver-dev=12.8.90=h5888daf_1
  - cuda-driver-dev_linux-64=12.8.90=h3f2d84a_1
  - cuda-gdb=12.8.90=h50b4baa_0
  - cuda-libraries=12.8.1=ha770c72_0
  - cuda-libraries-dev=12.8.1=ha770c72_0
  - cuda-libraries-static=12.8.1=ha770c72_0
  - cuda-nsight=12.8.90=h7938cbb_1
  - cuda-nvcc=12.8.93=hcdd1206_1
  - cuda-nvcc-dev_linux-64=12.8.93=he91c749_1
  - cuda-nvcc-impl=12.8.93=h85509e4_1
  - cuda-nvcc-tools=12.8.93=he02047a_1
  - cuda-nvcc_linux-64=12.8.93=h04802cd_1
  - cuda-nvdisasm=12.8.90=hbd13f7d_1
  - cuda-nvml-dev=12.8.90=hbd13f7d_0
  - cuda-nvprof=12.8.90=hbd13f7d_0
  - cuda-nvprune=12.8.90=hbd13f7d_1
  - cuda-nvrtc=12.8.93=h5888daf_1
  - cuda-nvrtc-dev=12.8.93=h5888daf_1
  - cuda-nvrtc-static=12.8.93=h5888daf_1
  - cuda-nvtx=12.8.90=hbd13f7d_0
  - cuda-nvvm-dev_linux-64=12.8.93=ha770c72_1
  - cuda-nvvm-impl=12.8.93=he02047a_1
  - cuda-nvvm-tools=12.8.93=he02047a_1
  - cuda-nvvp=12.8.93=hbd13f7d_1
  - cuda-opencl=12.8.90=hbd13f7d_0
  - cuda-opencl-dev=12.8.90=h5888daf_0
  - cuda-profiler-api=12.8.90=h7938cbb_1
  - cuda-sanitizer-api=12.8.93=hbd13f7d_1
  - cuda-toolkit=12.4.0=0
  - cuda-tools=12.8.1=ha770c72_0
  - cuda-version=12.8=h5d125a7_3
  - cuda-visual-tools=12.8.1=ha770c72_0
  - cxx-compiler=1.9.0=h1a2810e_0
  - dbus=1.13.6=h5008d03_3
  - expat=2.6.4=h5888daf_0
  - font-ttf-dejavu-sans-mono=2.37=hab24e00_0
  - font-ttf-inconsolata=3.000=h77eed37_0
  - font-ttf-source-code-pro=2.038=h77eed37_0
  - font-ttf-ubuntu=0.83=h77eed37_3
  - fontconfig=2.15.0=h7e30c49_1
  - fonts-conda-ecosystem=1=0
  - fonts-conda-forge=1=0
  - freetype=2.13.3=h48d6fc4_0
  - gcc=13.3.0=h9576a4e_2
  - gcc_impl_linux-64=13.3.0=h1e990d8_2
  - gcc_linux-64=13.3.0=hc28eda2_8
  - gds-tools=1.13.1.3=h5888daf_0
  - gmp=6.3.0=hac33072_2
  - gxx=13.3.0=h9576a4e_2
  - gxx_impl_linux-64=13.3.0=hae580e1_2
  - gxx_linux-64=13.3.0=h6834431_8
  - icu=75.1=he02047a_0
  - kernel-headers_linux-64=3.10.0=he073ed8_18
  - keyutils=1.6.1=h166bdaf_0
  - krb5=1.21.3=h659f571_0
  - ld_impl_linux-64=2.43=h712a8e2_4
  - libcap=2.75=h39aace5_0
  - libcublas=12.8.4.1=h9ab20c4_1
  - libcublas-dev=12.8.4.1=h9ab20c4_1
  - libcublas-static=12.8.4.1=h9ab20c4_1
  - libcufft=11.3.3.83=h5888daf_1
  - libcufft-dev=11.3.3.83=h5888daf_1
  - libcufft-static=11.3.3.83=h5888daf_1
  - libcufile=1.13.1.3=h12f29b5_0
  - libcufile-dev=1.13.1.3=h5888daf_0
  - libcufile-static=1.13.1.3=h5888daf_0
  - libcurand=10.3.9.90=h9ab20c4_1
  - libcurand-dev=10.3.9.90=h9ab20c4_1
  - libcurand-static=10.3.9.90=h9ab20c4_1
  - libcusolver=11.7.3.90=h9ab20c4_1
  - libcusolver-dev=11.7.3.90=h9ab20c4_1
  - libcusolver-static=11.7.3.90=h9ab20c4_1
  - libcusparse=12.5.8.93=hbd13f7d_0
  - libcusparse-dev=12.5.8.93=h5888daf_0
  - libcusparse-static=12.5.8.93=h5888daf_0
  - libedit=3.1.20250104=pl5321h7949ede_0
  - libexpat=2.6.4=h5888daf_0
  - libffi=3.4.6=h2dba641_0
  - libgcc=14.2.0=h767d61c_2
  - libgcc-devel_linux-64=13.3.0=hc03c837_102
  - libgcc-ng=14.2.0=h69a702a_2
  - libgcrypt-lib=1.11.0=hb9d3cd8_2
  - libglib=2.82.2=h2ff4ddf_1
  - libglvnd=1.7.0=ha4b6fd6_2
  - libgomp=14.2.0=h767d61c_2
  - libgpg-error=1.51=hbd13f7d_1
  - libiconv=1.18=h4ce23a2_1
  - liblzma=5.6.4=hb9d3cd8_0
  - libnl=3.11.0=hb9d3cd8_0
  - libnpp=12.3.3.100=h9ab20c4_1
  - libnpp-dev=12.3.3.100=h9ab20c4_1
  - libnpp-static=12.3.3.100=h9ab20c4_1
  - libnsl=2.0.1=hd590300_0
  - libnuma=2.0.18=h4ab18f5_2
  - libnvfatbin=12.8.90=hbd13f7d_0
  - libnvfatbin-dev=12.8.90=h5888daf_0
  - libnvfatbin-static=12.8.90=h5888daf_0
  - libnvjitlink=12.8.93=h5888daf_1
  - libnvjitlink-dev=12.8.93=h5888daf_1
  - libnvjitlink-static=12.8.93=h5888daf_1
  - libnvjpeg=12.3.5.92=h97fd463_0
  - libnvjpeg-dev=12.3.5.92=ha770c72_0
  - libnvjpeg-static=12.3.5.92=ha770c72_0
  - libopengl=1.7.0=ha4b6fd6_2
  - libpng=1.6.47=h943b412_0
  - libsanitizer=13.3.0=he8ea267_2
  - libsqlite=3.49.1=hee588c1_2
  - libstdcxx=14.2.0=h8f9b012_2
  - libstdcxx-devel_linux-64=13.3.0=hc03c837_102
  - libstdcxx-ng=14.2.0=h4852527_2
  - libsystemd0=257.4=h4e0b6ca_1
  - libudev1=257.4=hbe16f8c_1
  - libuuid=2.38.1=h0b41bf4_0
  - libxcb=1.17.0=h8a09558_0
  - libxcrypt=4.4.36=hd590300_1
  - libxkbcommon=1.8.1=hc4a0caf_0
  - libxkbfile=1.1.0=h166bdaf_1
  - libxml2=2.13.6=h8d12d68_0
  - libzlib=1.3.1=hb9d3cd8_2
  - lz4-c=1.10.0=h5888daf_1
  - ncurses=6.5=h2d0b736_3
  - nsight-compute=2025.1.1.2=hb5ebaad_0
  - nspr=4.36=h5888daf_0
  - nss=3.108=h159eef7_0
  - ocl-icd=2.3.2=hb9d3cd8_2
  - opencl-headers=2024.10.24=h5888daf_0
  - openssl=3.4.1=h7b32b05_0
  - pcre2=10.44=hba22ea6_2
  - pip=25.0.1=pyh8b19718_0
  - pthread-stubs=0.4=hb9d3cd8_1002
  - python=3.11.11=h9e4cc4f_2_cpython
  - rdma-core=56.0=h5888daf_0
  - readline=8.2=h8c095d6_2
  - sparsehash=2.0.4=hcb278e6_1
  - sysroot_linux-64=2.17=h0157908_18
  - tk=8.6.13=noxft_h4845f30_101
  - wayland=1.23.1=h3e06ad9_0
  - wheel=0.45.1=pyhd8ed1ab_1
  - xcb-util=0.4.1=hb711507_2
  - xcb-util-cursor=0.1.5=hb9d3cd8_0
  - xcb-util-image=0.4.0=hb711507_2
  - xcb-util-keysyms=0.4.1=hb711507_0
  - xcb-util-renderutil=0.3.10=hb711507_0
  - xcb-util-wm=0.4.2=hb711507_0
  - xkeyboard-config=2.43=hb9d3cd8_0
  - xorg-libice=1.1.2=hb9d3cd8_0
  - xorg-libsm=1.2.6=he73a12e_0
  - xorg-libx11=1.8.12=h4f16b4b_0
  - xorg-libxau=1.0.12=hb9d3cd8_0
  - xorg-libxcomposite=0.4.6=hb9d3cd8_2
  - xorg-libxdamage=1.1.6=hb9d3cd8_0
  - xorg-libxdmcp=1.1.5=hb9d3cd8_0
  - xorg-libxext=1.3.6=hb9d3cd8_0
  - xorg-libxfixes=6.0.1=hb9d3cd8_0
  - xorg-libxi=1.8.2=hb9d3cd8_0
  - xorg-libxrandr=1.5.4=hb9d3cd8_0
  - xorg-libxrender=0.9.12=hb9d3cd8_0
  - xorg-libxtst=1.2.5=hb9d3cd8_3
  - zstd=1.5.7=hb8e6e7a_2
  - pip:
      - addict==2.4.0
      - anyio==4.9.0
      - attrs==25.3.0
      - backports-cached-property==1.0.2
      - backports-tarfile==1.2.0
      - bbox==0.9.4
      - blessings==1.7
      - blinker==1.9.0
      - build==1.2.2.post1
      - cachecontrol==0.14.2
      - cachetools==5.5.2
      - certifi==2025.1.31
      - cffi==1.17.1
      - chardet==5.2.0
      - charset-normalizer==3.4.1
      - cleo==2.1.0
      - click==8.1.8
      - codecov==2.1.13
      - colorama==0.4.6
      - coloredlogs==15.0.1
      - colour-runner==0.1.1
      - configargparse==1.7
      - coverage==7.7.0
      - crashtest==0.4.1
      - cryptography==44.0.2
      - dash==3.0.0
      - deepdiff==8.4.2
      - distlib==0.3.9
      - dulwich==0.22.8
      - einops==0.8.1
      - fastjsonschema==2.21.1
      - filelock==3.18.0
      - findpython==0.6.3
      - flask==3.0.3
      - fsspec==2025.3.0
      - h11==0.14.0
      - httpcore==1.0.7
      - httpx==0.28.1
      - huggingface-hub==0.29.3
      - humanfriendly==10.0
      - idna==3.10
      - importlib-metadata==8.6.1
      - installer==0.7.0
      - itsdangerous==2.2.0
      - jaraco-classes==3.4.0
      - jaraco-context==6.0.1
      - jaraco-functools==4.1.0
      - jeepney==0.9.0
      - jinja2==3.1.6
      - joblib==1.4.2
      - jsonschema==4.23.0
      - jsonschema-specifications==2024.10.1
      - jupyter-core==5.7.2
      - keyring==25.6.0
      - markupsafe==3.0.2
      - more-itertools==10.6.0
      - mpmath==1.3.0
      - msgpack==1.1.0
      - narwhals==1.31.0
      - nbformat==5.10.4
      - nest-asyncio==1.6.0
      - networkx==3.4.2
      - numpy==1.26.4
      - nvidia-cublas-cu12==12.4.2.65
      - nvidia-cuda-cupti-cu12==12.4.99
      - nvidia-cuda-nvrtc-cu12==12.4.99
      - nvidia-cuda-runtime-cu12==12.4.99
      - nvidia-cudnn-cu12==9.1.0.70
      - nvidia-cufft-cu12==11.2.0.44
      - nvidia-curand-cu12==10.3.5.119
      - nvidia-cusolver-cu12==11.6.0.99
      - nvidia-cusparse-cu12==12.3.0.142
      - nvidia-nccl-cu12==2.20.5
      - nvidia-nvjitlink-cu12==12.4.99
      - nvidia-nvtx-cu12==12.4.99
      - open3d==0.18.0
      - orderly-set==5.3.0
      - packaging==24.2
      - pandas==2.2.3
      - pastel==0.2.1
      - pbs-installer==2025.3.17
      - pillow==11.1.0
      - pkginfo==1.12.1.2
      - platformdirs==4.3.7
      - plotly==6.0.1
      - pluggy==1.5.0
      - poethepoet==0.33.1
      - poetry==2.1.1
      - poetry-core==2.1.1
      - pyarrow==19.0.1
      - pycparser==2.22
      - pygments==2.19.1
      - pyproject-api==1.9.0
      - pyproject-hooks==1.2.0
      - pyquaternion==0.9.9
      - python-dateutil==2.9.0.post0
      - pytz==2025.1
      - pyyaml==6.0.2
      - rapidfuzz==3.12.2
      - referencing==0.36.2
      - regex==2024.11.6
      - requests==2.32.3
      - requests-toolbelt==1.0.0
      - rerun-sdk==0.22.1
      - retrying==1.3.4
      - rootpath==0.1.1
      - rpds-py==0.23.1
      - safetensors==0.4.5
      - scikit-learn==1.6.1
      - scipy==1.15.2
      - secretstorage==3.3.3
      - setuptools==77.0.3
      - shapely==2.0.7
      - shellingham==1.5.4
      - six==1.17.0
      - sniffio==1.3.1
      - spatiallm==0.0.1
      - stringcase==1.2.0
      - sympy==1.13.3
      - termcolor==2.5.0
      - terminaltables==3.1.10
      - threadpoolctl==3.6.0
      - tokenizers==0.20.3
      - toml==0.10.2
      - tomlkit==0.13.2
      - torch==2.4.1+cu124
      - torchaudio==2.4.1+cu124
      - torchsparse==2.1.0
      - torchvision==0.19.1+cu124
      - tox==4.24.2
      - tqdm==4.67.1
      - traitlets==5.14.3
      - transformers==4.46.1
      - triton==3.0.0
      - trove-classifiers==2025.3.19.19
      - typing-extensions==4.12.2
      - tzdata==2025.1
      - urllib3==2.3.0
      - virtualenv==20.29.3
      - werkzeug==3.0.6
      - zipp==3.21.0
      - zstandard==0.23.0
prefix: /root/anaconda3/envs/spatiallm

**Comment by SamMaoYS** [comment]
你好，我怀疑可能是生成的时候timeout了 你可以试着增大timeout的值 在这一行代码中 [line81](https://github.com/manycore-research/SpatialLM/blob/02d7e6e11db127483117440440805356c86bb222/inference.py#L81).

或者也可以尝试一下删除timeout，这样会取消timeout机制。

**Comment by 2152505** [comment]
Cool！解决了，谢谢！

![Image](https://github.com/user-attachments/assets/68e9d4e3-917c-42f2-812a-210fd63d5785)

**Comment by 2152505** [comment]
也就是实际原因可能是电脑配置不太够，跑得比较慢所以会造成timeout吧

**Comment by bertjiazheng** [comment]
Since the issue has already been resolved, I will proceed to close it.
